app/Transformers/ProductTransformer.php

```diff
@@ -335,4 +395,95 @@ private function removeColorOption(array $variant, ?int $colorPosition): array
             ...$updatedOptions,
         ];
     }
+
+    private function checkVariantMigrationEligibility(array $data, Collection $variants): bool
+    {
+        if(empty($data['product_links'])) {
+            return false;
+        }
+
+        if (!in_array(MigrationProduct::MATADOR_WEBSITE_ID, $data['extension_attributes']['website_ids'], true)) {
+            return false;
+        }
+
+        $needMigrate = true;
+        if (2 === $data['status']) {
+            $needMigrate = false;
+            foreach ($variants as $variant) {
+                if (1 === $variant->origin_data['status']) {
+                    $needMigrate = true;
+                    break;
+                }
+            }
+        }
+
+        if (!$needMigrate) {
+            return false;
+        }
+
+        return true;
```

Comment:
:large_blue_circle: this could be reduced to simpler conditions with early returns:

```suggestion
        if ($data['status'] != 2) {
            return false;
        }
        
        $hasEnabledVariants = collect($variants)
            ->some(fn ($item) => $item->origin_data['status'] === 1);
            
        return $hasEnabledVariants;
```

Reply:
we can't reject unactive grouped products because we can import them with active variants. in magento we can have grouped products disabled, but with active variants

Reply:
hmm, in that case, the logic you currently have is not working correctly. It will reject all grouped products that don't have any enabled variants.

Reply:
we can reconfirm with Hollie, but I understood that we don't migrate products without variants (or with disabled variants)

Reply:
Ok, I think I see the issue here. It's even more important to write it in a more readable way :)

```suggestion
        if ($data['status'] !== 2) {
            return true;
        }
        
        return collect($variants)
            ->some(fn ($item) => $item->origin_data['status'] === 1);
```

Does it make any sense?

Reply:
ok, i will update